core/src/test/java/com/datastax/oss/driver/internal/core/pool/ChannelPoolReconnectTest.java

```diff
@@ -133,4 +136,59 @@ public void should_reconnect_when_channel_starts_graceful_shutdown() throws Exce
 
     factoryHelper.verifyNoMoreCalls();
   }
+
+  @Test
+  public void should_let_current_attempt_complete_when_reconnecting_now()
```

Comment:
Ah yes, sorry forgot to mention when I approved.  I reran the suite repeatedly and could no longer reproduce the AssertionError on ths branch, where on 4.x it manifests ~60% of the time